server/app/services/agent/node_plugger.rb

```diff
@@ -31,7 +30,15 @@ def plugin!
     end
 
     def update_node
-      node.set(connected: true, last_seen_at: Time.now.utc)
+      result = HostNode.where(:id => node.id).any_of(
+        {:connected_at => nil}, {:connected_at.lt => @connected_at}
+      ).find_and_modify({:$set => {connected: true, last_seen_at: Time.now.utc, connected_at: @connected_at}}, {new: true})
+
+      if result
+        info "Connected node #{@node.to_path} at #{@connected_at}"
+      else
+        warn "Skip node #{@node.to_path} connect at #{@connected_at}, because node has re-connected at #{@node.connected_at}"
```

Comment:
This should probably also reject the websocket connection, but that is not possible to do right now.